test/parallel/test-runner-reporters.js

```diff
@@ -89,7 +89,7 @@ describe('node:test reporters', { concurrency: true }, () => {
                               ['--test', '--test-reporter', fixtures.path('test-runner/custom_reporters/', filename),
                                testFile]);
       assert.strictEqual(child.stderr.toString(), '');
-      assert.strictEqual(child.stdout.toString(), `${filename} {"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":7}`);
+      assert.match(child.stdout.toString(), new RegExp(`${filename} {"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":\\d}`));
```

Comment:
Passing a variable to `RegExp` constructor is a bit of a footgun, as future iteration of the test may introduce more regex special chars – to the point, currently it contains a dot which happens to make the test pass, but doesn't validate exactly what we meant. I suggest we split it in two checks to be safe.
```suggestion
      const stdout = child.stdout.toString(); 
      assert.match(stdout, /{"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":\d}$/);
      assert.strictEqual(stdout.slice(0, filename.length + 2), `${filename} {`);
```